simulators/ethereum/engine/engineclient.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/eth/catalyst"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/ethereum/hive/hivesim"
+)
+
+// Execution API v1.0.0-alpha.5 changes this to 8550
+// https://github.com/ethereum/execution-apis/releases/tag/v1.0.0-alpha.5
+var EnginePortHTTP = 8545
+var EnginePortWS = 8546
+
+// EngineClient wrapper for Ethereum Engine RPC for testing purposes.
+type EngineClient struct {
+	*hivesim.T
+	*hivesim.Client
+	http_c *rpc.Client
+	ws_c   *rpc.Client
+	c      *rpc.Client
+	Eth    *ethclient.Client
+	IP     net.IP
+
+	// This holds most recent context created by the Ctx method.
+	// Every time Ctx is called, it creates a new context with the default
+	// timeout and cancels the previous one.
+	lastCtx    context.Context
+	lastCancel context.CancelFunc
+}
+
+// NewClient creates a engine client that uses the given RPC client.
+func NewEngineClient(t *hivesim.T, hc *hivesim.Client) *EngineClient {
+	client := &http.Client{
+		Transport: &loggingRoundTrip{
+			t:     t,
+			hc:    hc,
+			inner: http.DefaultTransport,
+		},
+	}
+	// Prepare HTTP Client
+	rpcHttpClient, _ := rpc.DialHTTPWithClient(fmt.Sprintf("http://%v:%v/", hc.IP, EnginePortHTTP), client)
+	// Prepare WS Client
+	ctx, done := context.WithTimeout(context.Background(), 5*time.Second)
+	rpcWsClient, err := rpc.DialWebsocket(ctx, fmt.Sprintf("ws://%v:%v/", hc.IP, EnginePortWS), "")
```

Comment:
You would usually `defer done()` before the `DialWebsocket` call here, otherwise the dial will not timeout properly

Reply:
This part has been removed altogether .